src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
@spalger Replacing getIds with this simpler implementation is not going to be as clean as originally thought. [_index_pattern.js](https://github.com/elastic/kibana/blob/master/src/ui/public/index_patterns/_index_pattern.js#L352) calls `getIds.clearCache()` when an index pattern is created or saved. 

How should this be handled? Do index pattern ids even need to be cached?